pkg/controllers/resourceapply/config_hash_test.go

```diff
@@ -0,0 +1,271 @@
+package resourceapply
+
+import (
+	"context"
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	gmg "github.com/onsi/gomega"
+
+	configv1 "github.com/openshift/api/config/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/openshift/cluster-cloud-controller-manager-operator/pkg/cloud"
+	"github.com/openshift/cluster-cloud-controller-manager-operator/pkg/config"
+)
+
+func getOperatorConfigForAzureStack() config.OperatorConfig {
+	return config.OperatorConfig{
+		ManagedNamespace: "openshift-cloud-controller-manager",
+		ImagesReference: config.ImagesReference{
+			CloudControllerManagerOperator: "op",
+			CloudControllerManagerAzure:    "foo",
+			CloudNodeManagerAzure:          "bar",
+		},
+		PlatformStatus: &configv1.PlatformStatus{
+			Type: configv1.AzurePlatformType,
+			Azure: &configv1.AzurePlatformStatus{
+				CloudName: configv1.AzureStackCloud,
+			},
+		},
+		InfrastructureName: "my-cool-cluster-777",
+	}
+}
+
+func getVolumes() []corev1.Volume {
+	return []corev1.Volume{
+		{
+			Name: "foo", VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{SecretName: "secret"},
+			},
+		},
+		{
+			Name: "bar", VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{SecretName: "secret2"},
+			},
+		},
+		{
+			Name: "baz", VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{SecretName: "secret"},
```

Comment:
Deliberately sharing a name with volume foo?

Reply:
Yes

Reply:
I think a comment about why this is deliberately sharing the name would be useful